UPGRADE_GUIDE.md

```diff
@@ -5,6 +5,107 @@ To see discussion around these API changes, please refer to the
 [changelog](/CHANGELOG.md) and visit the commits and issues they
 reference.
 
+0.13.x -> 1.0.0
+----------------
+
+- React `0.14.x` is now required
```

Comment:
One thing I'm aware of is that context will come from parent instead of the owner. As far as I can tell, it shouldn't affect the router except for the AsyncProps module.